File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2: return n
        a, b, ans = 2, 1, 0
        for i in range(3, n+1):
            ans = a+b
            b = a
            a = ans
        return ans

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if(n <= 2) return n;
    let step1 = 2, step2 = 1, ans = 0;
    for(let i=2; i<n; i++){
        ans = step1 + step2;
        step2 = step1;
        step1 = ans;
    }
    return ans;
};